apps/src/lab2/views/dialogs/GenericConfirmationDialog.tsx

```diff
@@ -0,0 +1,47 @@
+import Typography from '@cdo/apps/componentLibrary/typography';
+import React from 'react';
+import {BaseDialogProps} from './DialogManager';
+import moduleStyles from './confirm-dialog.module.scss';
+import commonI18n from '@cdo/locale';
+
+/**
+ * Generic confirmation dialog used in Lab2 labs.
+ * The title, message, and confirm button text can be customized.
+ * If no confirm button text is provided, the default text is "OK" (translatable).
+ */
+const GenericConfirmationDialog: React.FunctionComponent<BaseDialogProps> = ({
+  handleConfirm,
+  handleCancel,
+  title,
+  message,
+  confirmText,
+}) => {
+  return (
+    <div className={moduleStyles.confirmDialog}>
+      <Typography semanticTag="h1" visualAppearance="heading-lg">
+        {title}
+      </Typography>
+      <Typography semanticTag="p" visualAppearance="body-two">
+        {message}
+      </Typography>
+      <div className={moduleStyles.buttonContainer}>
+        <button
+          className={moduleStyles.cancel}
+          type="button"
+          onClick={handleCancel}
+        >
+          {commonI18n.cancel()}
+        </button>
+        <button
```

Comment:
Question/thoughts for later (non-blocking)
- might be good to use design system buttons here? (as well as on the other lab2 dialogs like Start Over)
- @levadadenys do we have any plans for a generic dialog component in our design system? I'm wondering if a simple dialog with title, content, confirm and cancel actions is generic enough to be part of the design sytem.

Reply:
yeah we should probably update these to use the design system buttons, I filed a [task](https://codedotorg.atlassian.net/browse/CT-629) that is currently in the code tools backlog